letsencrypt/client.py

```diff
@@ -1,10 +1,12 @@
 """Let's Encrypt client API."""
+import functools
```

Comment:
Maybe `from functools import partial`?  Personally I find the fun of functional programming is gone if it's extremely verbose :)